addons/product_expiry/product_expiry.py

```diff
@@ -105,7 +105,7 @@ def _get_quants(self, cr, uid, ids, context=None):
 
     def apply_removal_strategy(self, cr, uid, location, product, qty, domain, removal_strategy, context=None):
         if removal_strategy == 'fefo':
-            order = 'removal_date, in_date, id'
+            order = 'removal_date, location_id, package_id, lot_id, in_date, id'
```

Comment:
@jco-odoo: I see @lbellier's point: in_date is not significant in FEFO, it's better to make stock workers' walk path shorter.